rerun_cpp/src/rerun/recording_stream.cpp

```diff
@@ -73,22 +73,19 @@ namespace rerun {
         const DataCell* data_cells
     ) {
         // Map to C API:
-        std::vector<rr_data_cell> c_data_cells;
-        c_data_cells.reserve(num_data_cells);
-        for (size_t i = 0; i < num_data_cells; ++i) {
-            c_data_cells.push_back({
-                .component_name = data_cells[i].component_name,
-                .num_bytes = static_cast<uint64_t>(data_cells[i].buffer->size()),
-                .bytes = data_cells[i].buffer->data(),
-            });
+        std::unique_ptr<rr_data_cell[]> c_data_cells =
```

Comment:
Why the use of a `unique_ptr` here? What was wrong with just having a vector on the stack and pointing to it?

Reply:
_afaik_ std::vector never has small array optimization, so it's a heap alloc regardless

Reply:
confirmed. std::vector can't implement small vector optimization because std::move on it isn't allowed to invalidate the iterator T_T
https://youtu.be/vElZc6zSIXM?t=233

Reply:
> don't need to reserve and doesn't have additional overhead for the counter. 

`std::vector<rr_data_cell> c_data_cells{num_data_cells};`

also has no reserve, and is arguably simpler

Reply:
That one is _technically_ slower since it zero-inits all elements. But yeah I guess that's the most appropriate one here!